pkg/tlsutil/tls.go

```diff
@@ -161,11 +165,15 @@ func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service
 	hasAppSecret := appSecret != nil
 	hasCASecretAndConfigMap := caSecret != nil && caConfigMap != nil
 	// TODO: handle passed in CA
+
```

Comment:
It appears to me that the code can be simpler if we just check if ca is passed in and pkg them into ca secret and config before we get the CA from the cluster using `getCASecretAndConfigMapInCluster` at line 160. Once you have done that, the rest of the code should take care of itself.

Reply:
yes but then we would have to call the code from 247 onwards in two locations since we would have to create the configMap  and secret in the cluster.

Reply:
if we have CA config and secret are created from the passed in CA, the `hasCASecretAndConfigMap` is always true. Then the code never hits the case `case: both CA and Application TLS assets don't exist.`. It will only be either the case `hasAppSecret && hasCASecretAndConfigMap` or `!hasAppSecret && hasCASecretAndConfigMap`.

Reply:
What I meant to say was if we have to create a CA config and secret from the passed in CA, the flow would be as follows:
1. Read the PEM bytes from the input file for Key and Cert
2. Parse PEM encoded bytes
3. Call `toCASecretAndConfigmap` to generate secret and ConfigMap
4. Call `scg.KubeClient.CoreV1().Secrets(ns).Create` to create secret and configMap

Since the above steps are already being done in the case where both CA and Application TLS assets don't exist, I thought it would be more clear if we combined the custom CA case with it.

Reply:
I thought that we should skip 4 for custom CA's?

Reply:
@rithujohn191 we can skip step 4. so that we don't put ca configmap and secret into the cluster.

Reply:
ah I see. Looks like this was finalized when I was out. Okay I will make the change

Reply:
I have made the changes according to the discussion